src/services/proof-of-good-smart-contract.service.ts

```diff
@@ -167,6 +167,34 @@ export class ProofOfGoodSmartContractService {
       return this.contract.setCap(activityId, oracleId, duration, points);
     });
   }
+
+  async getUserGoodPoints(userId: string) {
+    try {
+      const res = await this.contract.profileByUserId(userId);
```

Comment:
Given that I'm requesting at least one change, could we encapsulate this within the ethers poll try method to make it a bit more resilient ?

Reply:
@kharaone take a look at 048320fff9717861d877dc70897b6c0796f78c97 and let me know if my implementation should be modified... I'm not sure if the `eventName`/`events.find(...)` logic needs to be sustained for a read